app/charts/index.ts

```diff
@@ -685,31 +666,35 @@ const mkChartConfigAdjuster = ({
 }) => {
   // For filters & segments we can't reach a primitive level as we need to
   // pass the whole object.
-  const shouldAdjustConfigField = (configKey: string, configValue: unknown) =>
+  const shouldAdjustConfigField = (path: string, configValue: unknown) =>
     typeof configValue !== "object" ||
     Array.isArray(configValue) ||
-    ["filters", "segment"].includes(configKey);
+    ["filters", "fields.segment", "interactiveFiltersConfig.legend"].includes(
+      path
+    );
 
   const go = ({ path, field }: { path: string; field: Object }) => {
     for (const [k, v] of Object.entries(field)) {
       const newPath = path !== "" ? `${path}.${k}` : k;
 
-      if (shouldAdjustConfigField(k, v)) {
-        const adjustField: (params: FieldAdjustParams<ChartConfig>) => void =
-          get(chartAdjustConfig, newPath) ||
-          get(chartAdjustConfig, pathOverrides[newPath]);
-        adjustField?.({
-          oldValue: v,
-          newChartConfig,
-          oldChartConfig,
-          dimensions,
-          measures,
-        });
-      } else {
-        go({
-          path: newPath,
-          field: v,
-        });
+      if (v !== undefined) {
+        if (shouldAdjustConfigField(newPath, v)) {
```

Comment:
Nit: I just thought that maybe isConfigLeaf could be maybe more understandable ?